### ISSUE TITLE
Upgrade to Docusaurus v3

### ISSUE THREAD

**Issue by diwakergupta** [hirosystems/docs#566]
Not urgent, just filing for tracking.

Docusaurus v3 was [released last month](https://docusaurus.io/blog/releases/3.0) and v2 is now in maintenance mode. They recommend upgrading to v3 by January 2024.

The [upgrade guide](https://docusaurus.io/docs/migration/v3) seems mostly straight-forward.

I started making the changes just to get a sense of scope. Right now the primary blocker is the [openapi plugin](https://github.com/cloud-annotations/docusaurus-openapi), which does have an [approved (though not yet merged) PR](https://github.com/cloud-annotations/docusaurus-openapi/pull/266) for v3 support.

Once that is merged, hopefully this should be an easy lift.

**Comment by diwakergupta** [comment]
Quick update here: docusaurus has announced v2.x is no longer supported, released 3.0.1. The openapi plugin has also been updated, so that's no longer a blocker.

At this point, the only blocker is actual Markdown changes that break because of the switch from MDX1 to MDX3. This is in line with what they say: https://docusaurus.io/docs/migration/v3

> The main breaking change is the upgrade from MDX v1 to MDX v3

Specifically, a bunch of stuff in `_commands.md` now breaks due to the problems described here: https://docusaurus.io/docs/migration/v3#common-mdx-problems

Basically, bad usages of `{` and `<` that now need to be escaped properly.

Addressing this properly likely requires an update to stacks.js as well, I'll see if I can generate a PR easily. FYI @janniks
